src/hotspot/cpu/ppc/ppc.ad

```diff
@@ -11824,15 +11753,19 @@ instruct cmpF_reg_reg_Ex(flagsReg crx, regF src1, regF src2) %{
 %}
 
 // Compare float, generate -1,0,1
-instruct cmpF3_reg_reg_ExEx(iRegIdst dst, regF src1, regF src2) %{
+instruct cmpF3_reg_reg(iRegIdst dst, regF src1, regF src2) %{
   match(Set dst (CmpF3 src1 src2));
-  ins_cost(DEFAULT_COST*5+BRANCH_COST);
+  ins_cost(DEFAULT_COST * 6);
+  size(VM_Version::has_brw() ? 20 : 24);
 
-  expand %{
-    flagsReg tmp1;
-    cmpFUnordered_reg_reg(tmp1, src1, src2);
-    cmovI_conIvalueMinus1_conIvalue0_conIvalue1_Ex(dst, tmp1);
+  format %{ "cmpF3_reg_reg $dst, $src1, $src2" %}
+
+  ins_encode %{
+    __ fcmpu(CCR0, $src1$$FloatRegister, $src2$$FloatRegister);
+    __ cror(CCR0, Assembler::less, CCR0, Assembler::summary_overflow);
+    __ set_cmp3($dst$$Register);
```

Comment:
Why not
set_cmpu3($dst$$Register, true); // C2 requires unordered to get treated like less
?
(same for CmpD3)

Reply:
my bad, fixed! Thanks!